### ISSUE TITLE
feat: unique error IDs

### ISSUE THREAD

**Issue by dalvizu** [hashgraph/hedera-mirror-node#4643]
### Problem

While trying to trace https://github.com/hashgraph/hedera-mirror-node/issues/4642 and what might have gone wrong, I had to search a large number of logs, get some information from the original reporter, in order to understand why they may have received such a message:

![image](https://user-images.githubusercontent.com/731899/195212626-0a2e57c4-da28-4094-873e-83a543de2daa.png)

### Solution

It'dbegreatif this body included a unique error ID which can be easily searched for.

It'dbereallygreatif this error ID was used to log all messages like you can do with log4j's MDC capability: https://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/MDC.html

In a nutshell they store some ID in a threadlocal, clear that out and generate a new one when a web request comes in, and all logging for that statement is searchable with that ID. If there's an error you just write the ID and then that's how you action it when you get questions from your users about why they see such errors

### Alternatives

Trolling through a few handful of monitoring and logging systems and bugging Steven a lot.

**Comment by steven-sheehy** [comment]
We do have [request IDs](https://github.com/hashgraph/hedera-mirror-node/blob/3bbecbaaf39b6fedf1287975bcced71e8e850f82/hedera-mirror-rest/logger.js#L33) associated with requests in the REST API logs. We just don't return that to the client in any header or body. While we could expose those as a header (not in the body), I would prefer to have a standardized solution using [OpenTelemtry](https://opentelemetry.io/) and return span IDs.
